### PR TITLE
Upgrade compiler to Clang 9 and use engine libcxx on all platforms.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

**Comment by chinmaygarde** [comment]
The version of Clang does not have compiler `rt` support for `i386`. But, [we won't need that soon anyway](https://github.com/flutter/flutter/issues/22598#issuecomment-510371571). The linked issue is a Catalina blocker. Will land this after we no longer need an `i386` `gen_snapshot`.

File: build/config/BUILDCONFIG.gn [hunk 1]
```diff
@@ -535,7 +535,7 @@ if (custom_toolchain != "") {
 #
 # Variables
 #   no_default_deps: If true, no standard dependencies will be added.
-if (is_android || (is_linux && current_cpu != "x86")) {
+if (!is_win && current_cpu != "x86") {
```
Comment (gaaclarke) [inline build/config/BUILDCONFIG.gn]: This previously would pass for android of non x86, now it doesn't.  Was that intended?

Comment (chinmaygarde) [reply build/config/BUILDCONFIG.gn]: No. Fixing.

**Comment by chinmaygarde** [comment]
Updated patch incoming that only updates the compiler and not cxx.

**Pull request closed without merging**